src/main/java/com/targomo/client/api/response/TimeResponse.java

```diff
@@ -88,20 +94,39 @@ public void mapResults(final TravelOptions travelOptions, final JSONObject resul
 	 * @param source Source coordinate
 	 * @param target Target coordinate
 	 * @param travelTime Travel time to be added
+	 * @param length Travel distance to be added
 	 */
-	public void addTravelTime(Coordinate source, Coordinate target, Integer travelTime) {
+	public void addTravelWeight(Coordinate source, Coordinate target, Integer travelTime, Integer length) {
 		
-		this.travelTimes.putIfAbsent(source, new HashMap<>());
-		this.travelTimes.get(source).put(target, travelTime);
+		this.travelWeights.putIfAbsent(source, new HashMap<>());
+		this.travelWeights.get(source).put(target, new TravelWeight(length, travelTime));
 	}
-	
+
 	/**
 	 * @param source Source coordinate
 	 * @param target Target coordinate
 	 * @return null if the source or the target is not available, the travel time otherwise
 	 */
 	public Integer getTravelTime(Coordinate source, Coordinate target) {
-		return this.travelTimes.getOrDefault(source, null).getOrDefault(target, -1);
+		return this.getTravelWeight(source, target).getTravelTime();
+	}
+
+	/**
+	 * @param source Source coordinate
+	 * @param target Target coordinate
+	 * @return null if the source or the target is not available, the travel time otherwise
+	 */
+	public Integer getLength(Coordinate source, Coordinate target) {
+		return this.getTravelWeight(source, target).getTravelDistance();
+	}
+
+	/**
+	 * @param source Source coordinate
+	 * @param target Target coordinate
+	 * @return null if the source or the target is not available, the travel weight otherwise
+	 */
+	public TravelWeight getTravelWeight(Coordinate source, Coordinate target) {
+		return this.travelWeights.getOrDefault(source, null).getOrDefault(target, new TravelWeight(-1, -1));
```

Comment:
`return this.travelWeights.getOrDefault(source, Collections.emptyMap()).getOrDefault(target, new TravelWeight(-1, -1));` 

otherwise you might get an NPE

Reply:
also can you make `new TravelWeight(-1, -1)` a constant so we don't have to create it everytime